### PR TITLE
Feature/lu 5259 save validate

### PR DESCRIPTION
**Pull request by larissanne**

### CODE REVIEW

File: app/forms/view_form.py [hunk 1]
```diff
@@ -33,19 +33,16 @@ def internal_server_error(error):
...
     log.info("Request.form: %s", request.form)
-
     status_message = ""
     url_parameters = dict(
         zip(["survey", "period", "reference"], [inqcode, period, ruref]))
     parameters = build_uri(url_parameters)
-
-    if request.form and request.form['action'] == 'saveForm':
+    if request.form and request.form['action'] == 'saveandvalidate':
+        response_data = extract_responses(request.form)
```
Comment (RyanAldy) [inline app/forms/view_form.py]: Why has this now moved outside of the 'try' block?

File: app/templates/view_form/FormView.html [hunk 2]
```diff
@@ -97,12 +97,16 @@ <h1>{{ contributor_details['referencename'] }}</h1>
...
-                    <button class="btn btn--small" type="submit" id="saveFormButton" value="saveForm" name="action">
+                    <!--<button class="btn btn--small" type="submit" id="saveFormButton" value="saveForm" name="action">
                         <span class="btn__inner">Save</span>
                     </button>
 
                     <button class="btn btn--small" type="submit" value='validate' name='action'>
                         <span class="btn__inner">Validate</span>
+                    </button> -->
+
+                    <button class="btn btn--small" type="submit" value='saveandvalidate' name='action'>
```
Comment (RyanAldy) [inline app/templates/view_form/FormView.html]: value=saveandvalidate doesn't follow the naming standards.  I would go for either save-and-validate as this follows HTML attribute naming standards or saveAndValidate in line with JavaScript.  Either one is fine.

File: app/templates/view_form/FormView.html [hunk 3]
```diff
@@ -97,12 +97,16 @@ <h1>{{ contributor_details['referencename'] }}</h1>
                     <br>
                     <br>
 
-                    <button class="btn btn--small" type="submit" id="saveFormButton" value="saveForm" name="action">
+                    <!--<button class="btn btn--small" type="submit" id="saveFormButton" value="saveForm" name="action">
                         <span class="btn__inner">Save</span>
                     </button>
 
                     <button class="btn btn--small" type="submit" value='validate' name='action'>
                         <span class="btn__inner">Validate</span>
```
Comment (sufyaniqbal) [inline app/templates/view_form/FormView.html]: I would probably remove the commented code

File: app/templates/view_form/FormView.html [hunk 4]
```diff
@@ -97,12 +97,16 @@ <h1>{{ contributor_details['referencename'] }}</h1>
                     <br>
                     <br>
 
-                    <button class="btn btn--small" type="submit" id="saveFormButton" value="saveForm" name="action">
+                    <!--<button class="btn btn--small" type="submit" id="saveFormButton" value="saveForm" name="action">
```
Comment (sufyaniqbal) [inline app/templates/view_form/FormView.html]: I would remove the commented code

File: app/forms/view_form.py [hunk 5]
```diff
@@ -142,6 +126,19 @@ def view_form(inqcode, period, ruref):
...
         status_colour=status_colour)
 
+    return render_template(
+    template_name_or_list=form_view_template_HTML,
+    survey=inqcode,
+    period=period,
+    ruref=ruref,
+    data=response_and_validations,
+    override_button=override_button,
+    status_message=json.dumps(status_message),
```
Comment (sufyaniqbal) [inline app/forms/view_form.py]: are we keeping the status message for now?

File: app/forms/view_form.py [hunk 6]
```diff
@@ -41,47 +42,22 @@ def view_form(inqcode, period, ruref):
...
-        except ConnectionError as connection_error:
-            status_message = 'There was a problem with the connection ' + connection_error
-            log.info('Connection Error %s', connection_error)
-        except Timeout as timeout_error:
-            status_message = 'Timeout error ' + timeout_error
-            log.info('Timeout Error %s', timeout_error)
-        except RequestException as requests_error:
-            status_message = 'There was a problem with your request ' + requests_error + 'Please contact Take-On Support Team'
-            log.info('Requests Error: %s', requests_error)
+    # if request.form and request.form['action'] == 'saveForm':
```
Comment (RyanAldy) [inline app/forms/view_form.py]: May aswell remove this as it's no longer needed

File: app/forms/view_form.py [hunk 7]
```diff
@@ -7,6 +7,7 @@
 from app.utilities.combine_data import combine_responses_and_validations
 from app.utilities.check_status import check_status
 from app.setup import log, api_caller
+from app.utilities.combine_response_validations import combine_response_validations
```
Comment (RyanAldy) [inline app/forms/view_form.py]: This is the wrong version of the code...

current dev: from app.utilities.combine_data import combine_responses_and_validations, which then calls different code.  May want to merge changes of the most up to date dev first.

File: app/forms/view_form.py [hunk 8]
```diff
@@ -41,47 +42,22 @@ def view_form(inqcode, period, ruref):
...
     contributor_data = json.loads(contributor_details)
     validations = json.loads(validation_outputs)
+    status_message = ""
     status = contributor_data['data'][0]['status']
     status_colour = check_status(status)
 
     view_form_data = json.loads(view_forms)
 
-    response_and_validations = combine_responses_and_validations(view_form_data, filter_validations(validations))
+    response_and_validations = combine_response_validations(view_form_data, filter_validations(validations))
```
Comment (RyanAldy) [inline app/forms/view_form.py]: Wring version, code on the left is the correct code.

File: app/forms/view_form.py [hunk 9]
```diff
@@ -92,42 +68,31 @@ def view_form(inqcode, period, ruref):
...
-            validation=filter_validations(validations),
-            status_colour=status_colour)
+    contributor_data = json.loads(contributor_details)
+    validations = json.loads(validation_outputs)
+    status = contributor_data['data'][0]['status']
+    status_colour = check_status(status)
+
+    view_form_data = json.loads(view_forms)
+
+    response_and_validations = combine_responses_and_validations(view_form_data, filter_validations(validations))
```
Comment (RyanAldy) [inline app/forms/view_form.py]: It is used here too

File: app/forms/view_form.py [hunk 10]
```diff
@@ -171,3 +136,68 @@ def override_all_button(data):
         if validation_triggered_counter == 2:
             return True
     return False
+
+def save_form(parameters, requestform, inqcode, period, ruref):
```
Comment (RyanAldy) [inline app/forms/view_form.py]: Good separation into its own functions, however I would probably move them (save_form and validate) into its own file/module

File: app/forms/view_form.py (hunk 7 above)
Comment (RyanAldy) [inline app/forms/view_form.py]: Remove this, no longer used

**Pull request closed without merging**